Implement rainbow turntable effects

ASleepyCat opened the pull request:
3-in-1 PR, which implements static, reactive, and spinning rainbow turntable effects.

Also:
* Fix the regular spin effect not showing correctly for a split second when first switched to.
* Fix warning from `now-last_show < min_micros` comparison.

HWXLR8 commented:
Looks great! 2 concerns:

1. When cycling through the effects, `RAINBOW_REACT` is off until I move the TT, then it works as intended. 
2. The "reaction" is a bit slow imo, it would be cool to see a lot of movement with each scratch, rapid back-and-forth scratching does not show much of the effect. Same with `RAINBOW_SPIN`

ASleepyCat commented:
> When cycling through the effects, RAINBOW_REACT is off until I move the TT, then it works as intended.

Weird, I thought I fixed that bug. I've removed the optimisations so it should work now (they probably weren't needed anyway).

> The "reaction" is a bit slow imo

I've added `BEEF_TT_RAINBOW_SPIN_MUL` so that the user can adjust the speed of the spin effect.
